### ISSUE TITLE
QUESTION: Are there any examples of leveraging `alacritty-terminal` to do a custom text window (e.g. a rogue-like)?

### ISSUE THREAD

**Issue by Cthutu** [alacritty/alacritty#6479]
I am looking for a solution to be able to render unicode characters to a grid-based terminal window so I can implement various programs.  I want to be able to do something like crossterm/tui on a terminal window but use my own window and not a OS-provided one.

Alacritty-terminal seems like it's a fit but I cannot figure out how to set it up (do I need winit, how do I set up a proxy etc).  Are there any examples/documentation that can show how this is done?

I've looked at the alacritty source code and find it hard to follow and figure out a minimum application that can open a window and display "Hello, World" in it.

**Comment by Cthutu** [comment]
> That way people can actually pick whatever terminal emulator they like best. Anything else just seems backwards.

Not at all.  If I wanted to create a terminal looking program (text editor, rogue-like game) in a new window, or maybe support multiple windows connected to the same process, I could not do that in a terminal emulator.

I will look at `crossfont` but I also need the `grid` (I think) part of alacritty and windowing code to create a separate window that renders monospaced unicode font on a grid.  There is nothing in Rust that I can see that does this functionality along.  E.g. print a '*' character at this location on this window.

I was hoping alacritty-terminal did this, and that the `alacritty` crate used this for rendering an arbitrary window and its contents while dealing with all the terminal emulator integration.  It seems I was wrong.

The nearest I've seen to do the functionality I am looking for is `bracket-terminal` but it only deals with a 256 character monospaced font.

I will try to identify which part of alacritty creates a window, and renders the grid and try to extract that out.  Thanks for the tip about `crossfont`.  I didn't know about that crate.

**Comment by chrisduerr** [comment]
> Not at all. If I wanted to create a terminal looking program (text editor, rogue-like game) in a new window, or maybe support multiple windows connected to the same process, I could not do that in a terminal emulator.

Yes you can. That's what e-mail clients and tmux do for example using multiplexing.

I can pretty much guarantee you that whatever you're trying to do, you're doing it wrong.
